src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java

```diff
@@ -0,0 +1,184 @@
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static com.minecolonies.api.util.constant.TranslationConstants.COM_MINECOLONIES_COREMOD_GUI_BAKER;
+import static com.minecolonies.api.util.constant.TranslationConstants.COM_MINECOLONIES_COREMOD_GUI_WORKERHUTS_RETRIEVE_OFF;
+import static com.minecolonies.api.util.constant.TranslationConstants.COM_MINECOLONIES_COREMOD_GUI_WORKERHUTS_RETRIEVE_ON;
+
+/**
+ * Window for the fisherman hut.
+ */
+public class WindowHutLumberjack extends AbstractWindowWorkerBuilding<BuildingLumberjack.View>
+{
+    /**
+     * Id of the list in the pane.
+     */
+    private static final String LIST_SAPLINGS               = "trees";
+
+    /**
+     * Page the sapling selector is at.
+     */
+    private static final String PAGE_SAPLINGS               = "saplingActions";
+
+    /**
+     * Id of the button to change between true and false of the sapling.
+     */
+    private static final String BUTTON_CURRENT_SAPLING = "switch";
+
+    /**
+     * String describing on for the gui.
+     */
+    private static final String ON  = LanguageHandler.format(COM_MINECOLONIES_COREMOD_GUI_WORKERHUTS_RETRIEVE_ON);
+
+    /**
+     * String describing off for the gui.
+     */
+    private static final String OFF = LanguageHandler.format(COM_MINECOLONIES_COREMOD_GUI_WORKERHUTS_RETRIEVE_OFF);
+
+    /**
+     * Id of the pages view.
+     */
+    private static final String VIEW_PAGES                = "pages";
+
+    /**
+     * Scrolling list containing the saplings.
+     */
+    private       ScrollingList saplingsList;
+
+    /**
+     * List of saplings the lumberjack should, or should not fell (true if should, false if should not).
+     */
+    private final Map<ItemStack, Boolean> treesToFell = new LinkedHashMap<>();
+
+    /**
+     * The building of the lumberjack (Client side representation).
+     */
+    private final BuildingLumberjack.View ownBuilding;
+
+    /**
+     * Constructor for the window of the fisherman.
+     *
+     * @param building {@link BuildingBaker.View}.
+     */
+    public WindowHutLumberjack(final BuildingLumberjack.View building)
+    {
+        super(building, Constants.MOD_ID + ":gui/windowHutLumberjack.xml");
+        this.ownBuilding = building;
+        pullLevelsFromHut();
+    }
+
+
+    /**
+     * Retrieve levels from the building to display in GUI.
+     */
+    private void pullLevelsFromHut()
+    {
+        if (building.getColony().getBuilding(building.getID()) != null)
+        {
+            treesToFell.clear();
+            treesToFell.putAll(building.treesToFell);
+        }
+    }
+
+    @Override
+    public void onOpened()
+    {
+        super.onOpened();
+        saplingsList = findPaneOfTypeByID(LIST_SAPLINGS, ScrollingList.class);
+        saplingsList.setDataProvider(new ScrollingList.DataProvider()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 23 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)